ui/src/app/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -16,16 +17,20 @@ interface WorkflowLogsViewerProps {
 interface WorkflowLogsViewerState {
     error?: Error;
     lines: string[];
+    followLogStream: boolean;
 }
 
 export class WorkflowLogsViewer extends React.Component<WorkflowLogsViewerProps, WorkflowLogsViewerState> {
     constructor(props: WorkflowLogsViewerProps) {
         super(props);
-        this.state = {lines: []};
+        this.state = {
+            lines: [],
+            followLogStream: true
+        };
     }
 
     public componentDidMount(): void {
-        services.workflows.getContainerLogs(this.props.workflow, this.props.nodeId, this.props.container, this.props.archived).subscribe(
+        services.workflows.getContainerLogs(this.props.workflow, this.props.nodeId, this.props.container, this.props.archived, this.state.followLogStream).subscribe(
```

Comment:
I think it might be because this subscription is never closed or updated once the state is changed. I would use [`shouldComponentUpdate()`](https://reactjs.org/docs/react-component.html#shouldcomponentupdate) to listen for state changes, and closing the event stream if needed.